skiko/src/jvmMain/cpp/windows/directXRedrawer.cc

```diff
@@ -439,16 +447,22 @@ extern "C"
     JNIEXPORT jint JNICALL Java_org_jetbrains_skiko_redrawer_Direct3DRedrawer_getBufferIndex(
         JNIEnv *env, jobject redrawer, jlong devicePtr)
     {
-        DirectXDevice *d3dDevice = fromJavaPointer<DirectXDevice *>(devicePtr);
-        const UINT64 fenceValue = d3dDevice->fenceValues[d3dDevice->bufferIndex];
-        d3dDevice->bufferIndex = d3dDevice->swapChain->GetCurrentBackBufferIndex();
-        if (d3dDevice->fence->GetCompletedValue() < fenceValue)
-        {
-            GR_D3D_CALL_ERRCHECK(d3dDevice->fence->SetEventOnCompletion(fenceValue, d3dDevice->fenceEvent));
-            WaitForSingleObjectEx(d3dDevice->fenceEvent, INFINITE, FALSE);
+        __try {
+            DirectXDevice *d3dDevice = fromJavaPointer<DirectXDevice *>(devicePtr);
```

Comment:
We need to test it, code is pretty fragile.

Reply:
I think we should try to keep this code, and maybe we can collect some useful information about the problem if SEH occurs in this part of the code.
PS. In this part of code, I am unable to simulate SEH on my PC, but on my old laptop (with Intel HD 520), I sometimes encountered exceptions when resizing.